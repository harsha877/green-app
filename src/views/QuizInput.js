import * as React from 'react';
import { View, StyleSheet, Text, ScrollView, TextInput, Button } from 'react-native';
import { useState } from 'react';
import { writeCollection } from '../firebase/config';
import { checkLenghtEqualToZero } from '../utils/utils';
import { CUSTOM_QUESTION_HEAD_TAG, CUSTOM_QUESTION_NEXT_TAG, CUSTOM_QUESTION_SUBMIT_TAG, GREEN_DB_COLLECTION_QUIZES, QUESTION_TEMPLATE, QUIZ_TEMPLATE } from '../constant/constants';
import { Picker } from "@react-native-picker/picker";

import CustumeButton from "../components/button";

export default function QuizInput({ navigation, route }) {
  const [id, setId] = useState("");
  const [flag, setFlag] = useState(false);
  const [data, setData] = useState([]);
  const [question, setQuestion] = useState("");
  const [option1, setOption1] = useState("");
  const [option2, setOption2] = useState("");
  const [option3, setOption3] = useState("");
  const [option4, setOption4] = useState("");
  const [priority, setPriority] = useState(['select', '1', '2', '3', '4']);
  const [newpriority, setNewPriority] = useState(['0', '0', '0', '0']);
  const [textBoxSize, setTextBoxSize] = useState(40);
  const [quizTemplate, setquizTemplate] = useState({});
  const [temp, setTemp] = useState("temp");


  //var quizTemplate = '';
  function priorityCheck() {

    return newpriority.reduce((partial, a) => partial + parseInt(a), 0)

  }

  function handleNext() {

    if (!(priorityCheck() == 10)) {
      alert("Invalid priority order");
      return false;
    }
    if (checkLenghtEqualToZero(question) || checkLenghtEqualToZero(option1)
      || checkLenghtEqualToZero(option2) || checkLenghtEqualToZero(option3)
      || checkLenghtEqualToZero(option4)) {
      alert('Invalid Submission');
      return false;
    }
    //generation of document
    assignQuestion();
    //setData(data);
    setQuestion("");
    setOption1("");
    setOption2("");
    setOption3("");
    setOption4("");
    setNewPriority(['0', '0', '0', '0']);
    //console.log(quizTemplate);
    return true;
  }

  function assignQuestion() {

    let template = { ...QUESTION_TEMPLATE };
    template.question = question;
    template.option1 = option1;
    template.option2 = option2;
    template.option3 = option3;
    template.option4 = option4;
    template.weight = newpriority;
    let quizTemplateTemp = quizTemplate;
    quizTemplateTemp.questions.push(template);
    quizTemplateTemp.length = quizTemplateTemp.questions.length;
    setquizTemplate(quizTemplateTemp);
  }


  function handleSubmit() {
    if (handleNext() == false) {
      return;
    }
    var obj = { id };
    //data.unshift(obj);
    console.log(quizTemplate);

    writeCollection(quizTemplate, quizTemplate.quizName, GREEN_DB_COLLECTION_QUIZES)
      .then((response) => setquizTemplate({}))
      .catch((e) => console.log(e))

    setFlag(false);
    alert("Quiz submission is successfull")
    navigation.navigate('customer login');

    // setData([]);
    // setUser("");
  }

  const priorityChangeHandler = (item, question) => {
    let temPriority = newpriority, temPriority2;
    if (question == 'q1') {
      temPriority[0] = item;
    } else if (question == 'q2') {
      temPriority[1] = item;
    } else if (question == 'q3') {
      temPriority[2] = item;
    } else if (question == 'q4') {
      temPriority[3] = item;
    }
    setNewPriority(temPriority);
    setTemp(temp + " ")
  }
  const onSubmitQuizName = () => {
    if (checkLenghtEqualToZero(id)) {
      alert(" Name Canot be empty");
    } else {
      setFlag(true)
      let quizTemplateTemp = { ...QUIZ_TEMPLATE };
      quizTemplateTemp.quizName = id;
      quizTemplateTemp.customerID = route.params.user.username;
      quizTemplateTemp.customerName = route.params.user.name;
      quizTemplateTemp.searchKey = route.params.user.name +' | '+ id;
      quizTemplateTemp.questions = [];
      quizTemplateTemp.length = 0;
      setquizTemplate(quizTemplateTemp);
      //console.log(quizTemplateTemp);
    }

  }

  return (
    <ScrollView >
      {flag == false ? <>
        <TextInput style={{ ...styles.TextInput, height: textBoxSize }} placeholder="Enter Quiz Name" value={id}
          onChangeText={(text) => setId(text)}
          onContentSizeChange={(e) => setTextBoxSize(e.nativeEvent.contentSize.height)}
        />
        <CustumeButton style={styles.Button} onPressHandler={onSubmitQuizName} name='Submit' />
      </> : <>
        <Text style={styles.text}>{CUSTOM_QUESTION_HEAD_TAG}</Text>

        <TextInput multiline style={styles.TextInput} placeholder="Question" value={question} onChangeText={(text) => setQuestion(text)} />
        <TextInput style={styles.TextInput} placeholder="option 1" value={option1} onChangeText={(text) => setOption1(text)} />
        <View style={{ flexDirection: 'row' }} >
          <Text style={styles.label}>Weight</Text>
          <Picker onValueChange={(itemValue, itemIndex) => priorityChangeHandler(itemValue, 'q1')} style={styles.picker} selectedValue={newpriority[0]}>

            {priority.map(key => (
              <Picker.Item label={key} value={key} key={key} />
            ))}
          </Picker>
        </View>
        <TextInput style={styles.TextInput} placeholder="option 2" value={option2} onChangeText={(text) => setOption2(text)} />
        <View style={{ flexDirection: 'row' }} >
          <Text style={styles.label}>Weight</Text>
          <Picker onValueChange={(itemValue, itemIndex) => priorityChangeHandler(itemValue, 'q2')} style={styles.picker} selectedValue={newpriority[1]}>
            {priority.map(key => (
              <Picker.Item label={key} value={key} key={key} />
            ))}
          </Picker>
        </View>
        <TextInput style={styles.TextInput} placeholder="option 3" value={option3} onChangeText={(text) => setOption3(text)} />
        <View style={{ flexDirection: 'row' }} >
          <Text style={styles.label}>Weight</Text>
          <Picker onValueChange={(itemValue, itemIndex) => priorityChangeHandler(itemValue, 'q3')} style={styles.picker} selectedValue={newpriority[2]}>
            {priority.map(key => (
              <Picker.Item label={key} value={key} key={key} />
            ))}
          </Picker>
        </View>
        <TextInput style={styles.TextInput} placeholder="option 4" value={option4} onChangeText={(text) => setOption4(text)} />
        <View style={{ flexDirection: 'row' }} >
          <Text style={styles.label}>Weight</Text>
          <Picker onValueChange={(itemValue, itemIndex) => priorityChangeHandler(itemValue, 'q4')} style={styles.picker} selectedValue={newpriority[3]}>
            {priority.map(key => (
              <Picker.Item label={key} value={key} key={key} />
            ))}
          </Picker>
        </View>
        <View style={styles.footer}>
          <View style={styles.Button}><Button onPress={handleNext} title='NEXT' color="green" /></View>
          <View style={styles.Button}><Button onPress={handleSubmit} title='Submit' color="#841584" /></View>
        </View>

        <Text style={{ marginTop: 20 }}>{CUSTOM_QUESTION_NEXT_TAG}</Text>
        <Text> {CUSTOM_QUESTION_SUBMIT_TAG}</Text>
      </>
      }
    </ScrollView>
  )

}
const styles = StyleSheet.create({
  button1: {
    // display: "flex",
    // flexDirection: "column",
    // justifyContent: "center",
    // alignItems: "center",
    // marginTop: 15
  },

  TextInput: {
    //height: 40,
    margin: 15,
    //width: 250,
    borderWidth: 1,
    padding: 10,
    borderRadius: 10,
  },
  Button:
  {
    // height: 40,
    // width: 25,
    // marginTop: 5,
  },

  text: {
    height: 40,
    margin: 15,
    width: 250,
    color: 'red',
    fontWeight: 'bold',
    fontSize: 15,
    padding: 10,
  },

  label: {
    height: 40,
    margin: 15,
    padding: 10,
    margin: 10,
  },

  footer: {
    display: 'flex',
    flexDirection: 'row',
  },
  picker: {
    width: 73,
    //padding: 10,
    borderWidth: 1,
    borderColor: "vlack",

  },
});