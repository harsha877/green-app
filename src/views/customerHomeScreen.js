import React, { useState } from "react";
import { View, Text, Button, ScrollView, SafeAreaView, StyleSheet } from 'react-native';
import KeyboardAvoidingView from "react-native/Libraries/Components/Keyboard/KeyboardAvoidingView";

export default function CustomerHomeScreen({ navigation, onSubmitHandler }) {



    const handleCreateQuiz = () => {
        navigation.navigate('customer question');
    }

    const handleViewQuizzes = () =>{
        console.log("view quizzes");
    }

    return (
        <View style={styles.container}>
            <Text>Customer Homepage</Text>
            <Button  onPress={handleCreateQuiz} title='Create Quiz' />
            <Button  onPress={handleViewQuizzes} title='View Quizzes' />
            <Button  onPress={() => onSubmitHandler("")} title='Log Out' />
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
        alignItems: "center",
        marginTop: 30
    }
});


//{user == "" ? <CustomerLogin onSubmit={onSubmitHandler} /> : <QuizInput setUser={setUser} navigation = {navigation} />}