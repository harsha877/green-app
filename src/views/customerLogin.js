import * as React from 'react';
import { View, Text } from 'react-native';
// import bottomNav from './styles/bottomNav';

export default function CustomerLogin({ navigation }) {
    return (
        const [email, setEmail] = useState("");
         const [password, setPassword] = useState("");
 
   function handleLogin(e)
   {
      e.preventDefault();
      console.log(email);
      console.log(password);
   }
  return (
    <View style={styles.container}>

      <Image source={require('../assets/avatar.png')}/>
        <TextInput
        style={styles.TextInput}
          placeholder="Email"
          value={email}
          onChangeText={(email) => setEmail(email)}
        />

        <TextInput
        style={styles.TextInput}
          placeholder="Password"
          secureTextEntry={true}
          value={password}
          onChangeText={(password) => setPassword(password)}
        />
 
        <Button style={styles.Button} onPress={handleLogin} title='LOGIN'/>
    </View>
  );
}
 
const styles = StyleSheet.create({
  container: {
    display:"flex",
    flexDirection:"column",
    justifyContent:"center",
    alignItems:"center",
    marginTop:200
  },
 
  TextInput:  {
        height: 40,
        margin:15,
        width:250,
        borderWidth: 1,
        padding: 10,
      },

 
  Button:
  {
        height: 40,
        borderWidth: 1,
        marginTop:10,
        padding: 10,
  }
});
