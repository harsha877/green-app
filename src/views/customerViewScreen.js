import React, { useState } from "react";
import { View, Text, TextInput, ScrollView, StyleSheet } from 'react-native';
import { Button } from "react-native-web";
import CustumeButton from "../components/button";
import CustumeTable from "../components/table";
import { GREEN_DB_COLLECTION_QUIZES } from "../constant/constants";
import { searchQuizesForCustomer, writeCollection } from "../firebase/config";


export default function CustomerViewScreen({ navigation, route }) {

    const [user, setUser] = useState(route.params.user);
    const [quizzes, setQuizzes] = useState();
    const [quizFlag, setQuizFlag] = useState(true);
    const [optionsFlag, setOptionsFlag] = useState(false);
    const [editQuestionFlag, seteditQuestionFlag] = useState(false);
    const [selectedIndex, setSelectedIndex] = useState();
    const [questionIndex, setQuestionIndex] = useState();
    const [quiz, setQuiz] = useState();
    const [textBoxSize, setTextBoxSize] = useState(100);

    const handleViewQuizzes = (index) => {
        setSelectedIndex(index);
        setQuiz(quizzes[index])
        console.log("selected " + quizzes[index].quizName);
        setOptionsFlag(true);
        setQuizFlag(false);
        showQuestions();
    }

    const showQuestions = () => {
        if (optionsFlag && selectedIndex != undefined && quiz != undefined) {
            //console.log(quiz.questions.map(data => data));
            return (
                quiz.questions.map((data, index) =>
                    <Text style={styles.questions} onPress={() => setQuestionIndex(index)} key={index}>
                        {data.question + '\n'}
                    </Text>)
            );
        } else {
            console.log("view questions validations failed");
        }
    }


    const getData = () => {
        searchQuizesForCustomer(route.params.user.username).then(data => {
            setQuizzes(data);
            viewList();
            showQuestions();
        })
    }


    const viewList = () => {

        if (quizFlag && quizzes != undefined) {
            console.log("showing List of quizes")
            return (quizzes.map(
                (data, index) => {
                    //console.log(a);
                    return (
                        <CustumeButton key={index}
                            onPressHandler={() => handleViewQuizzes(index)}
                            name={data.quizName}
                            type='button' />
                    )
                }));
        }

    };

    const viewQuestions = () => {

        if (questionIndex != undefined) {
            return (
                <View style={styles.questionsEdit}>
                    <Text>Question: </Text>
                    <TextInput 
                        style={{ ...styles.textBoxes, height: textBoxSize }}
                        value={quiz.questions[questionIndex].question} 
                        onChangeText = {(question) => updatequestion("qestion",question)} 
                        multiline
                        onContentSizeChange={(e) => setTextBoxSize(e.nativeEvent.contentSize.height)}
                        />
                    <Text>Option 1: </Text>
                    <TextInput style= {styles.textBoxes} 
                        value={quiz.questions[questionIndex].option1}
                        onChangeText = {(question) => updatequestion("option1",question)} />
                    <Text>Option 2: </Text>
                    <TextInput style= {styles.textBoxes} 
                        value={quiz.questions[questionIndex].option2}
                        onChangeText = {(question) => updatequestion("option2",question)} />
                    <Text>Option 3: </Text>
                    <TextInput style= {styles.textBoxes} 
                        value={quiz.questions[questionIndex].option3}
                        onChangeText = {(question) => updatequestion("option3",question)}/>
                    <Text>Option 4: </Text>
                    <TextInput style= {styles.textBoxes} 
                        value={quiz.questions[questionIndex].option4}
                        onChangeText = {(question) => updatequestion("option4",question)}/>
                    <CustumeButton name="Save" onPressHandler={saveUpdatedData}/>
                </View>

            )
        }
    }

    const updatequestion = (topic ,text) => {
        let temp = {...quiz};
        if (topic === "qestion"){
            temp.questions[questionIndex].question = text;
        }else if (topic === "option1"){
            temp.questions[questionIndex].option1 = text;
        }else if (topic === "option2"){
            temp.questions[questionIndex].option2 = text;
        }else if (topic === "option3"){
            temp.questions[questionIndex].option3 = text;
        }else if (topic === "option4"){
            temp.questions[questionIndex].option4 = text;
        }
        setQuiz(temp);
    }

    const saveUpdatedData = () => {

        console.log(quiz.quizName+ " : data update");
        writeCollection(quiz, quiz.quizName, GREEN_DB_COLLECTION_QUIZES);
        alert("Changes Saved!");
        //to do need to uncomment according to requirment
        //navigation.navigate('customer login', {user});
    }

    const viewrefresh = () => {
        if (quizFlag) {
            return (<CustumeButton name="refresh" onPressHandler={getData} />)
        }
    }

    return (
        <ScrollView >
            <Text>Customer View page</Text>

            <View style={styles.list}>
                <CustumeButton name="refresh" onPressHandler={getData} type="button" />
                <Text>{viewList()}</Text>
            </View>
            <View style={styles.list}>{showQuestions()}</View>
            {viewQuestions()}

        </ScrollView>
    )
}


const styles = StyleSheet.create({
    container: {
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
        alignItems: "center",
        marginTop: 30
    },
    list: {
        justifyContent: 'space-between',
        flexDirection: 'column',
        justifyContent: "center",
        alignItems: "center",
    },
    questions: {
        //alignItems: "center",
        backgroundColor: '#1992d4',
        padding: 10,
        margin: 10,
        borderRadius: 10,
    },
    questionsEdit: {
        //alignItems: "center",
        backgroundColor: '#1992d4',
        padding: 10,
        margin: 10,
        borderRadius: 10,
    },
    textBoxes: {
        margin: 15,
        //width: 250,
        borderWidth: 1,
        padding: 10,
        borderRadius: 10,
    }
});


//{user == "" ? <CustomerLogin onSubmit={onSubmitHandler} /> : <QuizInput setUser={setUser} navigation = {navigation} />}