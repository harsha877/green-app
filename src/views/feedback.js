import * as React from 'react';
import { useState } from 'react';
import { Text, View, StyleSheet,ScrollView,Dimensions,TouchableOpacity,SafeAreaView,Image, TextInput,Button, Alert
 } from 'react-native';
import Constants from 'expo-constants';
import TypeWriter from "react-native-typewriter";
import RadioGroup, {RadioButtonProps} from 'react-native-radio-buttons-group';


const radioButtonsData = [{
    id: '1', // acts as primary key, should be unique and non-empty string
    label: 'Yes',
    value: 'Y'
    }, {
    id: '2',
    label: 'No',
    value: 'N'
    }]

export default function FeedBack({ route, navigation }) {
    const [text, onChangeText] = React.useState("");
    const [defaultRating,setdefaultRating] = useState(0)
    const [maxRating,setmaxRating]= useState([1,2,3,4,5])
    const [radioButtons, setRadioButtons] = useState(radioButtonsData)
            
    

    const RadioButtn = () => {
      
       function onPressRadioButton(radioButtonsArray) {
        setRadioButtons(radioButtonsArray);
      }
      return (
        <RadioGroup 
            radioButtons={radioButtons} 
            onPress={onPressRadioButton} 
        />
    );

    }

  const UseTextInput = (props) => {
  return (
    <TextInput
      {...props} // Inherit any props passed to it; e.g., multiline, numberOfLines below
      editable
      maxLength={40}
    />
  );
}

const UseTextInputMultiline = () => {
  const [value, onChangeText] = React.useState('');

  
  return (
    <View
      style={{
        backgroundColor: value,
        borderBottomColor: '#000000',
        borderWidth: 1,
      }}>
      <UseTextInput
        multiline
        numberOfLines={4}
        onChangeText={text => onChangeText(text)}
        value={value}
        style={{padding: 5}}
      />
    </View>
  );
}



    const CustomRatingBar = () => {
        return(
            <View style={styles.customRatingBarStyle}>
            {
              maxRating.map((item,key) => {
                return(
                  <TouchableOpacity
                  activeOpacity={0.7}
                  key={item}
                  onPress={() => setdefaultRating(item)}

                  >
                  <Image
                  style={styles.starImgStyle}
                  source={
                    item <= defaultRating
                    ? require("../../assets/star_filled.png")
                    : require("../../assets/star_corner.png")
                  }
                  />
                  </TouchableOpacity>
                )
              }

              )
            }
            </View>

        )
    }

    const data= [
      {
        "question": "Q1. How would you rate your overall experience?",
        "id": 1
      },
      {
        "question": "Q2. Was our App Useful?",
        "id":2
      },
      {
        "question": "Q3. Comments ",   
        "id":3 
      }
    ]

    const SubmitFeedBack = () =>{
        Alert.alert(
          "Feedback",
          "Your Feedback Submitted",
          [
        {
          text: "OK",
          onPress: () => navigation.goBack()
        }
      ]
        );

    }


    const FeedBackList = () => {
        return(
             <View style={styles.feedbackQuestionStyle}>
               {
                data.map((item,key) => {
                  return(
                    <TouchableOpacity key={key}>
                     <Text style={styles.Question}>
                       {item.question}
                     </Text>
                     {
                      item.id == 1
                     ?
                      <View>
                      <CustomRatingBar />  
                       <Text style={styles.textStyle}>
                        {defaultRating + '/' + maxRating.length}
                      </Text>
                      </View>
                     : 
                     item.id == 2
                     ?
                            <RadioButtn style={styles.radiostyle}/>
                     :
                            <UseTextInputMultiline/>                     
                     }
                  </TouchableOpacity>
                 
                  )
                   }
                )
        }
         </View>

     )
}

return (
  <SafeAreaView style={styles.container}>
  <ScrollView >
    <View style={styles.container}>
      <Text style={styles.text}>
              Rate App
      </Text>
      <FeedBackList/>
      
    </View>
    <View style={styles.subbutton}>
    <Button color={"#063f5c"}  title="Submit" onPress={SubmitFeedBack}  />
    </View>
  </ScrollView>
  </SafeAreaView>  
);

}






const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: "center",
		backgroundColor: "#ecf0f1",
		padding: 2,
	},
	paragraph: {
		borderColor: "black",
		borderWidth: 1,
		margin: 30,
		marginBottom: 3,
		borderRadius: 25,
		height: 55,
		backgroundColor: "#003f69",
		fontSize: 18,
		textAlign: "center",
		paddingHorizontal: 30,
	},
  text: {
		color: "#000000",
		fontSize: 20,
		fontWeight: "600",
    marginTop: 20
	},
	typeWriter: {
		color: "#2471A3",
		fontSize: 25,
		margin: 15,
		fontWeight: "500",
		textAlign: "center",
	},
	questionNo: {
		color: "#2471A3",
		fontSize: 18,
		margin: 5,
	},
	Question: {
		fontSize: 20,
		margin: 5,
		fontWeight: "500",
	},
	top: {},
	bottom: {
		position: "absolute",
		width: Dimensions.get("screen").width,
		bottom: 10,
	},
	box: {
		backgroundColor: "#2471A3",
		height: 80,
		marginTop: 110,
	},
	bottomWavy: {
		position: "absolute",
		bottom: 20,
	},
  customRatingBarStyle: {
    justifyContent: 'center',
    flexDirection: 'row',
    marginTop: 30
  },
  feedbackQuestionStyle: {
    justifyContent: 'center',
    flexDirection: 'column',
  },
  starImgStyle:{
    width: 40,
    height: 40,
    resizeMode: 'cover'
  },
  radiostyle:{
    justifyContent: 'center',
    flexDirection: 'column'
  },
  input: {
    height: 40,
    margin: 12,
    borderWidth: 1,
    padding: 10,
  },
  subbutton: {
   
    fontsize: 30,
    margin:10,
    padding: 10,
    marginTop: 20,
		borderRadius: 50
   },
  textStyle: {
    textAlign: 'center',
    fontSize: 23,
    marginTop: 20
  }
})
