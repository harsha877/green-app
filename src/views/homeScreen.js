import React, {useState} from 'react';
import {View, Text, StyleSheet, ScrollView} from 'react-native';
import TypeWriter from 'react-native-typewriter';
import Questions from '../json/questions.json';
import Option from '../components/options.js';
import CustumeButton from '../components/button';

export default function HomeScreen({route, navigation}) {
  const {index} = route.params;
  return (
    <ScrollView>
      <TypeWriter style={styles.typeWriter} typing={1}>
        Water Footprint Calculator
      </TypeWriter>
      <View style={styles.container}>
        <Text style={styles.questionNo}>
          {index + 1}/{Questions.questions.length}
        </Text>
        <Text style={styles.Question}>
          {Questions.questions[index].question}
        </Text>
        {Questions.questions[index].answers.map(user => {
          let keys = Object.keys(user);
          return (
            <Text style={styles.paragraph}>
              {keys.map(key => {
                return (
                  <Option
                    value={`${key}`}
                    optionIdx={`${user[key]}`}
                    navigation={navigation}
                    qnIndex={index}
                  />
                );
              })}
            </Text>
          );
        })}
      </View>
    </ScrollView>
  );
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: '#ecf0f1',
    padding: 2,
  },
  paragraph: {
    borderColor: 'black',
    borderWidth: 1,
    margin: 20,
    marginBottom: 3,
    borderRadius: 20,
    // height: 65,
    padding: 20,
    backgroundColor: '#003f69',
    textAlign: 'center',
    color: '#000000',
    justifyContent: 'center',
    alignItems: 'center',
    alignContent: 'center',
    fontSize: 18,
  },
  typeWriter: {
    color: '#409cd0',
    fontSize: 25,
    margin: 15,
    fontWeight: '500',
    textAlign: 'center',
  },
  questionNo: {
    color: '#409cd0',
    fontSize: 18,
    margin: 5,
  },
  Question: {
    fontSize: 18,
    margin: 5,
  },
});
