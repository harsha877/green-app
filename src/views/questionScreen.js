import React, { useState } from "react";
import { View, Text, StyleSheet } from 'react-native';
import CustumeButton from "../components/button";
import { USER_CUSTOM_QUIZ_HOMESCREEN, USER_CUSTOM_QUIZ_RESULTSCREEN } from "../constant/constants";

export default function QuestionScreen({ navigation }) {

    const RsultPage = () => {
        
        navigation.navigate(USER_CUSTOM_QUIZ_RESULTSCREEN);
        
    };

    return (
        <View>
            <View>
                <Text>
                    question 1
                </Text>
            </View>
            <View>
                <CustumeButton name = 'Option 1' />
                <CustumeButton name = 'Option 2' />
                <CustumeButton name = 'Option 3' />
                <CustumeButton name = 'Option 4' />
            </View>
            <View style = {styles.navigation}>
                <CustumeButton name = 'Previous' />
                <CustumeButton name = 'Next' />
                <CustumeButton name = 'Submit' onPressHandler = {RsultPage} />
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    options:{

    },
    navigation:{
        justifyContent: 'space-between',
        flexDirection: 'row'
    },
})
