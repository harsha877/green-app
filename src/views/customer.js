import React, { useState } from "react";
import { View, Text, Button, ScrollView, SafeAreaView, StyleSheet } from 'react-native';
import KeyboardAvoidingView from "react-native/Libraries/Components/Keyboard/KeyboardAvoidingView";
import CustomerHomeScreen from "./customerHomeScreen";
import CustomerLogin from "./customerLogin";
import QuizInput from "./QuizInput";
export default function Customer({ navigation }) {
    const [user, setUser] = useState("");

    const onSubmitHandler = (data) => {
        setUser(data);
    };

    return (
        <KeyboardAvoidingView style={styles.container}>
            {user == "" ? <CustomerLogin onSubmit={onSubmitHandler} /> : <CustomerHomeScreen onSubmitHandler = {onSubmitHandler} navigation = {navigation} />}
        </KeyboardAvoidingView>
    )
}

const styles = StyleSheet.create({
    container: {
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
        alignItems: "center",
        marginTop: 30
    }
});