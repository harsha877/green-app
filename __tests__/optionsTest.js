import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import Options from '../src/components/options';
import { useStateValue } from "../src/views/stateProvider"

var arr = { value: 1, key: 1, optionIdx: 1, navigation: { NavigationContainer }, qnIndex: 0 }
test("Options Component Loading", () => {
    render(<Options prop={arr} />)

})

test("Option component snapshot", () => {
    let component = render(<Options prop={arr} />).toJSON();
    expect(component).toMatchSnapshot();
})


describe("Testing Option component", () => {
    let component;
    beforeEach(() => {
        component = render(<Options prop={arr} />);

    })

    test("Testing Styles", () => {
        const props = component.toJSON();
        //console.log(props);
        let Styles = props.children[0].props.style;
        //console.log(Styles);
        expect(Styles).toStrictEqual({
            opacity: 1,
        });
    })

    test("Testing Styles text", () => {
        const props = component.toJSON();
        console.log(props);
        let Styles = props.children[0].children[0].props.style;
        console.log(Styles);
        expect(Styles).toStrictEqual({
            fontSize: 20,
            color: "white",
            textAlign: "center",
        });
    })

    /*
    test("Testing Button Text",()=>{
        const {getAllByText}=ButtonComponent;
        expect(getAllByText("welcome").length).toBe(1);})

    test("Testing Styles",()=>{
        const props=ButtonComponent.toJSON();
        let Styles=props.children[0].children[0].props.style;
        expect(Styles).toStrictEqual({
            color: "#ffffff",
            fontSize: 18,
            textAlign: "center",
            padding: 5,
        });})*/



});