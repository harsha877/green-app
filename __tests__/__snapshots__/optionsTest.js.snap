// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option component snapshot 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
